### ISSUE TITLE
PHP 8.2 not supported

### ISSUE THREAD

**Issue by flashlabs** [PHP-CS-Fixer/PHP-CS-Fixer#6764]
### Description

PHP CS Fixer is not working with PHP `8.2.*`

### Runtime version

```
PHP needs to be a minimum version of PHP 7.4.0 and maximum version of PHP 8.1.*.
Current PHP version: 8.2.1.
To ignore this requirement please set `PHP_CS_FIXER_IGNORE_ENV`.
If you use PHP version higher than supported, you may experience code modified in a wrong way.
Please report such cases at https://github.com/PHP-CS-Fixer/PHP-CS-Fixer .
```

### Used command

```
vendor/bin/php-cs-fixer
```

### Configuration file

```
<?php

declare(strict_types=1);

$finder = (new PhpCsFixer\Finder())
    ->in(__DIR__.'/src')
    ->in(__DIR__.'/tests')
    ->in(__DIR__.'/config')
    ->in(__DIR__.'/migrations')
;

return (new PhpCsFixer\Config())
    ->setRules([
        '@PHP82Migration' => true,
        '@Symfony' => true,
        '@Symfony:risky' => true,
        'array_syntax' => ['syntax' => 'short'],
        'blank_line_before_statement' => [
            'statements' => ['break', 'continue', 'declare', 'exit', 'goto', 'return', 'throw', 'try'],
        ],
        'declare_strict_types' => true,
        'no_php4_constructor' => true,
        'no_useless_concat_operator' => true,
        'no_useless_else' => true,
        'no_useless_return' => true,
        'phpdoc_order' => true,
        'phpdoc_order_by_value' => true,
        'strict_comparison' => true,
        'strict_param' => true,
    ])
    ->setRiskyAllowed(true)
    ->setFinder($finder)
;
```

### Code snippet that reproduces the problem

n/a

**Comment by no-simpler** [comment]
I’m also surprised that PHP 8.2 is not yet supported, but I don’t think it’s a bug: there still are [open issues](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues?q=is%3Aissue+is%3Aopen+label%3Atopic%2FPHP8.2+) on this topic.

Last time I searched, `php-cs-fixer` does not explicitly declare the supported PHP versions in its documentation. However, `composer.json` states `"php": "^7.4 || ^8.0",` which should probably be changed to `"php": "^7.4 || 8.0.* || 8.1.*",`.

**Comment by flashlabs** [comment]
> I’m also surprised that PHP 8.2 is not yet supported, but I don’t think it’s a bug: there still are [open issues](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues?q=is%3Aissue+is%3Aopen+label%3Atopic%2FPHP8.2+) on this topic.

This was the best category I could choose from the suggested ones in the "New issue" form.

I know that there's no explicit declaration of supported versions, but since there's a stable `PHP` `8.2` I was expecting `php-cs-fixer` to work with that one.

There's also another blocker regarding the `php-cs-fixer`, which is `"doctrine/annotations": "^1.14.1"` and annotations are already in `2.*` version, but that's a different story 😅 

Anyway, fingers crossed for quick fixes on those bugs you mentioned 🤞

**Comment by keradus** [comment]
Folks, highly encouraging to use the search feature.
TL;DR - 8,2 is not currently supported as it takes huge effort to support each new PHP syntax. If you need it, consider to support the project [eg with code contribution].
ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6714#issuecomment-1359166470

**Comment by bocharsky-bw** [comment]
@keradus is there any simple way to know which PHP version is currently supported by PHP CS Fixer? I don't see this info in the main README file, what do you think about adding this? I think it would help to avoid most (some) reported issues about unsupported PHP version

**Comment by bocharsky-bw** [comment]
@keradus Thanks! How about #6768 ?

**Comment by keradus** [comment]
Let's move the discussion to PR then, thanks for picking up!

**Comment by flashlabs** [comment]
TBH the simplest and fastest way is to look at the `composer.json`
```
"php": "^7.4 || ^8.0",
```
but information in `README` might be useful ofc.

**Comment by dozer111** [comment]
@keradus have u a milestone, or sth like this about what tasks(goals) must be closed to support 8.2?

**Comment by keradus** [comment]
This is what we identified so far, but not sure if all is identified already
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues?q=is%3Aissue+is%3Aopen+8.2+label%3Atopic%2FPHP8.2

**Comment by stof** [comment]
@keradus as all those issues are not closed, is there any reason not to mark PHP 8.2 as supported ?

**Comment by Wirone** [comment]
@stof what do you mean? PHP 8.2 **is** supported. Error described in the [initial comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6764#issue-1556990218) currently is [visible for PHP 8.3](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/173db343ed768a722b9eb0968a1d0ed052a7475c/php-cs-fixer#L31).

**Comment by stof** [comment]
ah sorry, I missed the fact that the async-aws project was not using the latest version
